src/EditorFeatures/Core/Implementation/AutomaticCompletion/IEditorBraceCompletionSession.cs

```diff
@@ -2,19 +2,19 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-#nullable disable
-
 using System.Threading;
+using Microsoft.CodeAnalysis.BraceCompletion;
 using Microsoft.CodeAnalysis.Host;
 using Microsoft.VisualStudio.Text.BraceCompletion;
 
 namespace Microsoft.CodeAnalysis.Editor.Implementation.AutomaticCompletion
 {
     internal interface IEditorBraceCompletionSession : ILanguageService
     {
-        bool CheckOpeningPoint(IBraceCompletionSession session, CancellationToken cancellationToken);
-        void AfterStart(IBraceCompletionSession session, CancellationToken cancellationToken);
+        bool IsValidForBraceCompletion(char brace, int openingPosition, Document document, CancellationToken cancellationToken);
+        BraceCompletionResult? GetBraceCompletion(IBraceCompletionSession session, CancellationToken cancellationToken);
+        BraceCompletionResult? GetChangesAfterCompletion(IBraceCompletionSession session, CancellationToken cancellationToken);
         bool AllowOverType(IBraceCompletionSession session, CancellationToken cancellationToken);
-        void AfterReturn(IBraceCompletionSession session, CancellationToken cancellationToken);
+        BraceCompletionResult? GetChangesAfterReturn(IBraceCompletionSession session, CancellationToken cancellationToken);
```

Comment:
group these method better (i.e. put the GetChangesXXX methods together).  Please doc the distinction and semantics as this is not obvious.  i.e. what's the diff from GetBraceCompletion and GetChangesAfterCompletion.  why do both return the same type?  (i honestly don't know/understand).